adapters/repos/db/inverted/searcher.go

```diff
@@ -191,6 +192,9 @@ func (s *Searcher) extractPropValuePair(filter *filters.Clause,
 		out.children = make([]*propValuePair, len(filter.Operands))
 
 		eg := errgroup.Group{}
+		// prevent unbounded concurrency, see
+		// https://github.com/weaviate/weaviate/issues/3179 for details
+		eg.SetLimit(2 * runtime.GOMAXPROCS(0))
```

Comment:
Is `SetLimit(2 * runtime.GOMAXPROCS(0))` necessary now whenever `errgroup.Group` is used, to prevent regression on #3179? Or just for specific cases?